src/mbgl/sprite/sprite_parser.cpp

```diff
@@ -150,6 +152,36 @@ std::optional<style::ImageContent> getContent(const JSValue& value, const char*
     return std::nullopt;
 }
 
+std::optional<style::TextFit> parseTextFit(const std::string& value) {
+    static const std::unordered_map<std::string, style::TextFit> textFitMap = {
+        {"stretchOrShrink", style::TextFit::stretchOrShrink},
+        {"stretchOnly", style::TextFit::stretchOnly},
+        {"proportional", style::TextFit::proportional}
+    };
+
+    auto it = textFitMap.find(value);
+    if (it != textFitMap.end()) {
+        return it->second;
+    } else {
+        return std::nullopt;
+    }
+}
+
+std::optional<style::TextFit> getTextFit(const JSValue& value, const char* property, const char* name) {
+    if (value.HasMember(property)) {
+        auto& v = value[property];
+        if (v.IsString()) {
+            return parseTextFit(v.GetString());
```

Comment:
This would create std::string and copy underlying buffer. ML/N already uses std::string_view and I think it will be a good fit here.